includes/class-wporg-gp-translation-events-active-events-cache.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+class WPORG_GP_Translation_Events_Active_Events_Cache_Entry {
+	public int $event_id;
+	public DateTimeImmutable $start;
+	public DateTimeImmutable $end;
+
+	public function __construct( int $event_id, DateTimeImmutable $start, DateTimeImmutable $end ) {
+		$this->event_id = $event_id;
+		$this->start    = $start;
+		$this->end      = $end;
+	}
+}
+
+class WPORG_GP_Translation_Events_Active_Events_Cache {
+	public const CACHE_DURATION = 60 * 60 * 24; // 24 hours.
+	private const KEY = 'translation-events-active-events';
+	private static bool $invalidated_on_event_save = false;
+
+	public function __construct() {
+		// Invalidate cache when events are modified.
+		add_filter(
```

Comment:
This should be moved out of the constructor because if there are multiple instances of this class, we'll be listening to the hook multiple times.